### PR TITLE
B-19837: TRDM Malformed Data Handling

### PR DESCRIPTION
**Pull request by TevinAdams**
When TRDM receives malformed data  then the LOA and TAC parsers need to check for that and handle accordingly. In the context of the PR, malformed data is when a pipe delimited file is received that has a row or rows that are missing a column. 

Before this PR:
When TRDM Lambda function received a pipe delimited file with TAC rows that are missing a column then the TAC parser would fail with index out of bounds. LOA parser already accounted for this case and would handle the situation accordingly.

After this PR:
When TRDM Lambda function receives a pipe delimited file with TAC rows that are missing a column then the TAC parser will not fail but instead not parse that row as it is malformed.

Tests Added:
Added tests for LOA parser and TAC parser  for the case of receiving malformed data.

### CODE REVIEW

File: src/main/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParser.java [hunk 1]
```diff
@@ -88,6 +93,13 @@ public List<TransportationAccountingCode> parse(byte[] fileContent, XMLGregorian
 
     private TransportationAccountingCode processLineIntoTAC(String[] values, Map<String, Integer> columnHeaders,
             XMLGregorianCalendar trdmLastUpdate) throws RuntimeException {
+
+         // Check if value length does not align with columns
+         if (values.length != columnHeaders.size()) {
+            logger.info("TGET file row is malfomred. This row of TAC data will not be parsed.");
+            return null; // Skip this line
+        }
```
Comment (cameroncaci) [inline src/main/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParser.java:101]: Typo on "malfomred"

File: src/test/java/com/milmove/trdmlambda/milmove/util/LineOfAccountingParserTest.java [hunk 2]
```diff
@@ -97,7 +97,26 @@ void testLoaParser() throws IOException, DatatypeConfigurationException {
         assertEquals(expectedCode, result.get(0));
     }
 
-        private LocalDateTime convertXMLGregorianCalendarToLocalDateTime(XMLGregorianCalendar xmlGregorianCalendar) {
+    // Test to make sure the parser does not fail when recieinvg malfomred pipe data and does not add the malformed data to the codes array that will be inserted into the DB
+    @Test
```
Comment (cameroncaci) [inline src/test/java/com/milmove/trdmlambda/milmove/util/LineOfAccountingParserTest.java:101]: Typo on "malfomred"

File: src/test/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParserTest.java [hunk 3]
```diff
@@ -64,6 +64,26 @@ void testTacParser() throws IOException, DatatypeConfigurationException {
         assertEquals(expectedCode, result.get(0));
     }
 
+     // Test to make sure the parser does not fail when recieinvg malfomred pipe data and does not add the malformed data to the codes array that will be inserted into the DB
+    @Test
```
Comment (cameroncaci) [inline src/test/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParserTest.java:68]: Typo on "malfomred"

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Meets AC and tests work like a charm. Just spotted some typos

File: src/main/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParser.java (hunk 1 above)
Comment (TevinAdams) [reply src/main/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParser.java]: @cameroncaci  Fixed

File: src/test/java/com/milmove/trdmlambda/milmove/util/LineOfAccountingParserTest.java (hunk 2 above)
Comment (TevinAdams) [reply src/test/java/com/milmove/trdmlambda/milmove/util/LineOfAccountingParserTest.java]: @cameroncaci  Fixed

File: src/test/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParserTest.java (hunk 3 above)
Comment (TevinAdams) [reply src/test/java/com/milmove/trdmlambda/milmove/util/TransportationAccountingCodeParserTest.java]: @cameroncaci Fixed

**Review by deandreJones — APPROVED** [review]

**Review by cameroncaci — APPROVED** [review]

**Pull request merged**